modules/search-service-impl/src/main/java/org/opencastproject/search/endpoint/SearchRestService.java

```diff
@@ -433,6 +446,39 @@ public Response getEpisodes(
     return Response.ok(gson.toJson(json)).build();
   }
 
+  /**
+   * Iterate recursively through Object List and sign all Strings with key=url
+   * @param obj
+   */
+  private void findURLsAndSign(Object obj) {
+    if (obj instanceof Map) {
+      Map<String, Object> map = (Map<String, Object>) obj;
+      for (Map.Entry<String, Object> entry : map.entrySet()) {
+        if (entry.getKey().equals("url") && entry.getValue() instanceof String) {
+          String urlToSign = (String) entry.getValue();
+          if (this.urlSigningService.accepts(urlToSign)) {
+            try {
+              String signedUrl = this.urlSigningService.sign(
+                  urlToSign,
+                  expireSeconds,
+                  null,
+                  null);
+              map.put(entry.getKey(), signedUrl);
+            } catch (UrlSigningException e) {
+              logger.debug("Unable to sign url '" + urlToSign + "'.");
```

Comment:
I've overlooked this. You should usually avoid using string concatenation in log statements. Use the built-in placeholders instead:
```suggestion
              logger.debug("Unable to sign url '{}'.", urlToSign);
```